src/status_im/ui/screens/add_new/new_chat/views.cljs

```diff
@@ -61,9 +64,29 @@
     :yourself
     (i18n/label :t/can-not-add-yourself)))
 
+(defn filter-contacts [filter-text contacts]
+  (let [lower-filter-text (string/lower-case (str filter-text))
+        filter-fn         (fn [{:keys [name alias nickname]}]
+                            (or
+                             (string/includes? (string/lower-case (str name)) lower-filter-text)
+                             (string/includes? (string/lower-case (str alias)) lower-filter-text)
+                             (when nickname
+                               (string/includes? (string/lower-case (str nickname)) lower-filter-text))))]
+    (if filter-text
+      (filter filter-fn contacts)
+      contacts)))
+
+(defn is-valid-username? [username]
+  (let [is-chat-key? (and (string? username)
+                          (string/starts-with? username "0x")
```

Comment:
`(string/starts-with? username "0x")` we probably already have some helper function for this

Reply:
Found one in `status-im.transport.filters.core`, thanks!